psm_utils/io/timscore.py

```diff
@@ -0,0 +1,98 @@
+"""Reader for TIMScore Parquet files."""
+
+import logging
+import re
+from pathlib import Path
+from typing import Union
+
+import numpy as np
+import pandas as pd
+
+from psm_utils import PSM
+from psm_utils.io._base_classes import ReaderBase
+from psm_utils.peptidoform import format_number_as_string
+
+logger = logging.getLogger(__name__)
+
+
+class TIMScoreReader(ReaderBase):
+    """Reader for TIMScore Parquet files."""
+
+    def __init__(
+        self,
+        filename: Union[str, Path],
+        *args,
+        **kwargs,
+    ) -> None:
+        """
+        Reader for TIMScore Parquet files.
+
+        Parameters
+        ----------
+        filename: str, pathlib.Path
+            Path to MSF file.
+
+        """
+        super().__init__(filename, *args, **kwargs)
+        self._decoy_pattern = re.compile(r"^Reverse_")
+
+        self.data = pd.read_parquet(self.filename)
+
+    def __len__(self):
+        """Return number of PSMs in file."""
+        return len(self.data)
+
+    def __iter__(self):
+        """Iterate over file and return PSMs one-by-one."""
+        for entry in self.data.itertuples():
+            yield PSM(
+                peptidoform=_parse_peptidoform(
+                    entry.stripped_peptide, entry.ptms, entry.ptm_locations, entry.precursor_charge
+                ),
+                spectrum_id=entry.ms2_id,
+                is_decoy=all(self._decoy_pattern.match(p) for p in entry.locus_name),
+                score=entry.tims_score,
+                precursor_mz=entry.precursor_mz,
+                retention_time=entry.rt,
+                ion_mobility=entry.ook0,
+                protein_list=list(entry.locus_name),
+                rank=entry.rank,
+                source="TIMScore",
+                provenance_data={
+                    "candidate_id": str(entry.candidate_id),
+                    "ms2_id": str(entry.ms2_id),
+                    "parent_id": str(entry.parent_id),
+                },
+                metadata={
+                    "leading_aa": str(entry.leading_aa),
+                    "trailing_aa": str(entry.trailing_aa),
+                    "corrected_ook0": str(entry.corrected_ook0),
```

Comment:
i would recommend using the ook0 directly instead of corrected_ook0.  Not all psms will have caorrected_ook0